src/deepsparse/loggers/helpers.py

```diff
@@ -50,8 +50,52 @@ def possibly_extract_value(value: Any, remainder: Optional[str] = None) -> Any:
     if not remainder:
         return value
 
-    for sub_remainders in remainder.split("."):
-        value = value.__getattribute__(sub_remainders)
+    # splits remainder into separate strings.
+    # each string can access the new nesting depth and optionally
+    # do indexing and slicing at this depth
+    for sub_remainder in remainder.split("."):
+        square_brackets = re.search(r"\[(.*?)\]", sub_remainder)
+        if square_brackets:
+            # retrieve the string without the square brackets
+            sub_remainder = sub_remainder.split("[")[0]
```

Comment:
what do we do with the rest here?

Reply:
nothing. the actual body of the square brackets gets extracted using regex